src/components/Co2Calculator.jsx

```diff
@@ -0,0 +1,253 @@
+import styled from "styled-components";
+import { useEffect, useState } from "react";
+import { saveToLocal, loadFromLocal } from "../library/localStorage";
+import NumberInput from "./NumberInput";
+/* import isProductValid from "../library/validation"; */
+
+function Co2Calculator() {
+  const initialValues = {
+    inputCar: 0,
+    inputTrain: 0,
+    inputFlight: 0,
+  };
+
+  const [values, setValues] = useState(initialValues);
+  const [hasFormErrors, setHasFormErrors] = useState(false);
+
+  const handleInputChange = (name, value) => {
+    setValues({
+      ...values,
+      [name]: value,
+    });
+  };
+
+  const initialCalculationObject = {
+    carKm: 0,
+    trainKm: 0,
+    flightKm: 0,
+    co2Car: 0,
+    co2Train: 0,
+    co2Flight: 0,
+    co2BudgetSpent: 0,
+    co2BudgetLeft: 1500,
+  };
+
+  const localStorageCo2Calculation = loadFromLocal("co2calculation");
+  const [co2Calculation, setCo2Calculation] = useState(
+    localStorageCo2Calculation ?? initialCalculationObject
+  );
+  useEffect(() => {
+    saveToLocal("co2calculation", co2Calculation);
+  }, [co2Calculation]);
+
+  const handleSubmit = (event) => {
+    event.preventDefault();
+    if (values.inputCar > 0) {
+      const totalCarKm = co2Calculation.carKm + values.inputCar;
+      const co2Car = (co2Calculation.carKm + values.inputCar) * 0.119;
+      const co2CarRounded = Math.round(co2Car * 100) / 100;
+      const co2BudgetSpent =
+        Math.round(
+          (co2Calculation.co2BudgetSpent + values.inputCar * 0.119) * 100
+        ) / 100;
+      const co2BudgetLeft =
+        Math.round(
+          (co2Calculation.co2BudgetLeft - values.inputCar * 0.119) * 100
+        ) / 100;
+      setCo2Calculation({
+        ...co2Calculation,
+        carKm: totalCarKm,
+        co2Car: co2CarRounded,
+        co2BudgetSpent: co2BudgetSpent,
+        co2BudgetLeft: co2BudgetLeft,
+      });
+      setValues(initialValues);
+      setHasFormErrors(false);
```

Comment:
If inputCar is greater than 0, the handleSubmit will only run this part.
So if there are also values for inputTrain or inputFlight they will get lost.

Maybe you can avoid this with a conditional operator.
``` 
values.inputCar > 0 && { function }
values.inputTrain > 0 && { function }
values.inputFlight > 0 && { function }
```